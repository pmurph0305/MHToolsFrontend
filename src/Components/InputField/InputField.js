import React from 'react';

const InputField = ({onClick, placeholder}) => {
    let textInput = React.createRef();
    function handleClick() {
        onClick(textInput.current);
    }

    return (
        <div className="pa2">
            <input 
                ref={textInput}
                placeholder={placeholder}
                id="newtask" 
                type="text" 
                label="New task text input"
                aria-label="New task text input"
                className="f6 f5-l input-reset fl black-80 bg-white pa2 lh-solid w-100 w-75-m w-80-l br2-ns br--left-ns"
            />
            <button 
                value="add_task"
                onClick={handleClick}
                 //onClick={() => onClick(document.getElementById("newtask").value)}
                className="f6 f5-l babutton-reset fl pv2 tc bn bg-animate bg-black hover-bg-black-70 white pointer w-100 w-25-m w-20-l br2-ns br--right-ns"
            >Add New Task</button>
        </div>
    )
}

export default InputField;