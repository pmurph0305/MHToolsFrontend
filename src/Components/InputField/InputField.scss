@import '../../styles/shared.scss';

.InputContainer {
    padding: 0.5rem;
}

.InputField {
    padding: 0.5rem;
    background-color: white;
    color: $primary-text-color;
    // Large screens
    @media screen and (min-width:60em) {
        width: 80%;
    }
    // Medium
    @media screen and (min-width:30em) and (max-width: 60em) {
        width: 75%;
    }
    // Small
    @media screen and (max-width: 30em) {
       width: 100%;
    }
}

.AddTaskButton {
    @extend %ButtonBase;
    float: right;
    // Large
    @media screen and (min-width:60em) {
        width: 20%;
    }
    // Medium
    @media screen and (min-width:30em) and (max-width: 60em) {
        width: 25%;
    }
    // Small
    @media screen and (max-width: 30em) {
       width: 100%;
    }
}