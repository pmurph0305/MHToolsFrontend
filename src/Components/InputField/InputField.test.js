import { shallow, mount } from "enzyme";
import React from "react";
import InputField from "./InputField";

describe("InputField tests", () => {
  let wrapper;
  const mockClick = jest.fn(val => val);
  let mockProps = {
    onClick: mockClick,
    buttonTitle: "test button",
    placeholder: "test"
  };

  beforeEach(() => {
    wrapper = shallow(<InputField {...mockProps} />);
  });

  // test if exists.
  it("Exists", () => {
    expect(wrapper.exists()).toBe(true);
  });

  // snapshot test.
  it("Matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("Displays placeholder correctly", () => {
    expect(wrapper.find("input").props().placeholder).toBe("test");
  });

  // mount & simulate a click, make sure the return value is empty as nothing is entered.
  it("Calls on click when button clicked and returns value", () => {
    const wrapmount = mount(<InputField {...mockProps} />);
    wrapmount.find("button").simulate("click");
    expect(mockClick.mock.results[0].value.value).toBe("");
    expect(mockClick).toHaveBeenCalledTimes(1);
  });
});
