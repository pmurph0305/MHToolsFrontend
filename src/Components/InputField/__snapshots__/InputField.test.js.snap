// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField tests Matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputField
    onClick={[MockFunction]}
    placeholder="test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          aria-label="New task text input"
          className="InputField"
          id="newtask"
          label="New task text input"
          placeholder="test"
          type="text"
        />,
        <button
          className="AddTaskButton"
          onClick={[Function]}
          value="add_task"
        >
          Add New Task
        </button>,
      ],
      "className": "InputContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-label": "New task text input",
          "className": "InputField",
          "id": "newtask",
          "label": "New task text input",
          "placeholder": "test",
          "type": "text",
        },
        "ref": Object {
          "current": null,
        },
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add New Task",
          "className": "AddTaskButton",
          "onClick": [Function],
          "value": "add_task",
        },
        "ref": null,
        "rendered": "Add New Task",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            aria-label="New task text input"
            className="InputField"
            id="newtask"
            label="New task text input"
            placeholder="test"
            type="text"
          />,
          <button
            className="AddTaskButton"
            onClick={[Function]}
            value="add_task"
          >
            Add New Task
          </button>,
        ],
        "className": "InputContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-label": "New task text input",
            "className": "InputField",
            "id": "newtask",
            "label": "New task text input",
            "placeholder": "test",
            "type": "text",
          },
          "ref": Object {
            "current": null,
          },
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add New Task",
            "className": "AddTaskButton",
            "onClick": [Function],
            "value": "add_task",
          },
          "ref": null,
          "rendered": "Add New Task",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
